tests/functional/workloads/cnv/test_vm_snapshot_cloning_ops.py

```diff
@@ -51,30 +97,123 @@ def test_vm_clone(
         )
         vm_list = vm_objs_def + vm_objs_aggr
         log.info(f"Total VMs to process: {len(vm_list)}")
-        for index, vm_obj in enumerate(vm_list):
+        failed_vms = []
+        for vm_obj in vm_list:
+            # Expand PVC if `pvc_expand_before_clone` is True
+            pvc_obj = vm_obj.get_vm_pvc_obj()
+            if pvc_expand_before_clone:
+                new_size = 50
+                try:
+                    pvc_obj.resize_pvc(new_size=new_size, verify=True)
+                    pvc_obj = vm_obj.get_vm_pvc_obj()
+
+                    # Get rootdisk name
+                    disk = (
+                        vm_obj.vmi_obj.get()
+                        .get("status")
+                        .get("volumeStatus")[1]["target"]
+                    )
+                    devicename = f"/dev/{disk}"
+
+                    result = vm_obj.run_ssh_cmd(
+                        command=f"lsblk -d -n -o SIZE {devicename}"
+                    ).strip()
```

Comment:
The process of resizing the PVC and verifying it is repeated twice (once before cloning and once after). Please create a common function outside the test and reuse it the test

Reply:
Done